pkg/kubectl/cmd/delete.go

```diff
@@ -223,7 +231,7 @@ func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
 			// To preserve backwards compatibility, but prevent accidental data loss, we convert --grace-period=0
 			// into --grace-period=1 and wait until the object is successfully deleted. Users may provide --force
 			// to bypass this wait.
-			o.WaitForDeletion = true
+			o.WaitForGracefulDeletion = true
```

Comment:
Now that am adding wait to reapers in this PR, we can remove that GET based waiting logic.
The UID based waiting logic wont work for 1.7 kubectl with 1.6 apiserver, so am keeping it as is.
Will remove the GET based waiting logic in 1.8 (will update this PR if it misses 1.7)